src/lib/flow/sol-flow-resolver-conffile.c

```diff
@@ -142,19 +169,19 @@ _resolver_conffile_get_module(const char *type)
     SOL_NULL_CHECK(name, NULL);
 
     /* the hash entry keys are the type part only */
-    entry = g_hash_table_lookup(resolver_conffile_dlopens, name);
+    entry = find_entry_by_name(&resolver_conffile_dlopens, name);
     if (entry) {
         free(name);
         goto found;
     }
 
-    entry = calloc(1, sizeof(*entry));
-    if (!entry) {
-        SOL_DBG("Could not alloc memory for entry");
-        free(name);
-        return NULL;
+    if (resolver_conffile_dlopens.len == 0) {
+        atexit(resolver_conffile_clear_data);
```

Comment:
man atexit() says that if registered multiple times, it is called once per registration. Then do not use atexit() here, rather do once at the beginning.